### ISSUE TITLE
SvelteKit: New search results indicator

### ISSUE THREAD

**Issue by jasonhawkharris** [sourcegraph/sourcegraph#61030]
Design has produced a much better experience for displaying macro statistics for a given search. View the design [here](https://www.figma.com/file/EPgl0qVbqGSumvf52brhda/Code-Search?type=design&node-id=218-6499&mode=design&t=neu9YXpRcumG8PaG-0). This should be implemented in the svelte web app.

**Comment by jasonhawkharris** [comment]
Requesting input from a backend engineer (@camdencheek @bahrmichael @BolajiOlajide) because there are a few use cases where we do not have corresponding backend responses. 

This is what design has come up with for each eventuality, accompanied by the issue with each.
1. The search results are fetching 
    - One issue is that the search duration is not returned from the backend while fetching. Should we run a timer on the client side and 
    - ![Image](https://github.com/sourcegraph/sourcegraph/assets/62355966/f37372d9-3b87-4129-a9e4-5b40a288b2ed)
2. Search finished because it satisfied a certain constraint for stopping the search (I.e. display limit hit). The Backend has a sufficient response for this use case. 
3. The search is taking too long
    - How do we determine whether a search is taking too long? Does this require a backend response? Or is it okay to accomplish this on the client side?
    - ![Image](https://github.com/sourcegraph/sourcegraph/assets/62355966/9a64e4f8-3bec-4d00-8807-c6b3cef01d3e)
4. The search completes without issue. (The backend has a sufficient response to this use case)
5. There was an error while running the search. (The backend has a sufficient response to this use case)

In short, we need a backend response that returns when a search is taking too long that is similar in format to the responses served by successful search, constraint satisfaction, or error cases. As of now, this does not exist.

**Comment by jasonhawkharris** [comment]
[This file](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@52cfcf907f00ced9e11642bf94ccf0771abcfa89/-/blob/internal/search/streaming/api/progress.go) contains what is returned for each use case.

**Comment by bahrmichael** [comment]
I think solving (1) and (3) in the frontend is good.

For (1): The number reflects the user's experience to me. If my internet is slow and needs 5 seconds to establish a connection (grr German train internet) that's what I experience, and therefore the backend response (which doesn't know about my time to establish a connection) would be wrong. I think this is the frontend's job.

For (3): I think we can set a constant on the frontend after which we show that it's taking too long. Thinking more about this I can't come up with a reason why the backend would need to make decision if it's taking too long on every request (at least for showing this info). Maybe at some point this info might be configured via a feature flag, but starting with a constant value in the frontend sounds like a good approach to me.

**Comment by camdencheek** [comment]
> The number reflects the user's experience to me.

Yeah...this has historically been measured on the backend, but it's kinda misleading. The unfortunate thing is, like mentioned, a user's connection can be the source of noise, which means we're taking some responsibility for things out of our control. I'm in favor of just measuring on the client though -- makes it less confusing when we say "search completed in 0.02 seconds" when it was very obviously slower than that.

**Comment by BolajiOlajide** [comment]
> I think solving (1) and (3) in the frontend is good.
> 
> For (1): The number reflects the user's experience to me. If my internet is slow and needs 5 seconds to establish a connection (grr German train internet) that's what I experience, and therefore the backend response (which doesn't know about my time to establish a connection) would be wrong. I think this is the frontend's job.
> 
> For (3): I think we can set a constant on the frontend after which we show that it's taking too long. Thinking more about this I can't come up with a reason why the backend would need to make decision if it's taking too long on every request (at least for showing this info). Maybe at some point this info might be configured via a feature flag, but starting with a constant value in the frontend sounds like a good approach to me.

I agree with Michael's response for (3) above, I wonder if we can use data to  come up with an acceptable threshold for when we figure a search response is taking too long.

For (1), we could instead show how much has elapsed for searching instead of trying to figure out how long the search will take (which is a bit tricky to tell) as per Camden's point above.

**Comment by camdencheek** [comment]
Closing as done in #61033 (not sure why automation didn't close this 🤔)
